docs/index.rst

```diff
@@ -58,6 +58,10 @@ Welcome to Hera's documentation!
    examples/workflow_with_custom_image
    examples/workflow_with_template_ref
 
+.. toctree::
+   :caption: Expr Transpilation
+
+   expr
```

Comment:
This is a positive feedback comment :) 

At first I thought of suggesting we [place these docs with the other ones](https://github.com/argoproj-labs/hera-workflows/blob/main/docs/index.rst) but I think adding `expr` as its own section in the docs allows a fairly complicated topic to be isolated into its own logical unit of explanation, which users will likely appreciate